core/node/core.go

```diff
@@ -47,25 +48,38 @@ func BlockService(cfg *config.Config) func(lc fx.Lifecycle, bs blockstore.Blocks
 }
 
 // Pinning creates new pinner which tells GC which blocks should be kept
-func Pinning(bstore blockstore.Blockstore, ds format.DAGService, repo repo.Repo) (pin.Pinner, error) {
-	rootDS := repo.Datastore()
+func Pinning(reprovidingStrategy string) func(bstore blockstore.Blockstore, ds format.DAGService, repo repo.Repo, prov provider.System) (pin.Pinner, error) {
```

Comment:
We cannot use the fx-provided reprovidingStrategy here, as that is only set when the reprovider is created, and the reprovider needs the pinner.